std/algorithm/sorting.d

```diff
@@ -869,31 +873,54 @@ template multiSort(less...) //if (less.length > 1)
             alias ss = SwapStrategy.unstable;
             alias funs = less;
         }
-        alias lessFun = binaryFun!(funs[0]);
 
-        static if (funs.length > 1)
+        static if (funs.length == 1)
+            return sort!fun[0](r);
+        else
         {
-            while (r.length > 1)
+            void multiSortImpl(funs...)(Range r) @safe
             {
-                auto p = getPivot!lessFun(r);
-                auto t = partition3!(less[0], ss)(r, r[p]);
-                if (t[0].length <= t[2].length)
+                alias lessFun = binaryFun!(funs[0]);
+
+                static if (funs.length > 1)
                 {
-                    .multiSort!less(t[0]);
-                    .multiSort!(less[1 .. $])(t[1]);
-                    r = t[2];
+                    while (r.length > 1)
+                    {
+                        auto p = getPivot!lessFun(r);
+                        auto t = partition3!(funs[0], ss)(r, r[p]);
+                        if (t[0].length <= t[2].length)
+                        {
+                            multiSortImpl!funs(t[0]);
+                            multiSortImpl!(funs[1 .. $])(t[1]);
+                            r = t[2];
+                        }
+                        else
+                        {
+                            multiSortImpl!(funs[1 .. $])(t[1]);
+                            multiSortImpl!funs(t[2]);
+                            r = t[0];
+                        }
+                    }
                 }
                 else
                 {
-                    .multiSort!(less[1 .. $])(t[1]);
-                    .multiSort!less(t[2]);
-                    r = t[0];
+                    sort!(lessFun, ss)(r);
                 }
             }
-        }
-        else
-        {
-            sort!(lessFun, ss)(r);
+
+            static auto predFun(ElementType!Range a, ElementType!Range b)
+            {
+                foreach (f; AliasSeq!(funs))
+                {
+                    alias lessFun = binaryFun!(f);
+                    int r = lessFun(a, b);
```

Comment:
This will break code, there is no warranty  that lessFun returns `int`. Sometime it may return floating point, long or user defined type.
So the best way is
```d
if (auto r = lessFun(a, b))
    return r;
```